src/m_menu.c

```diff
@@ -5945,7 +5945,7 @@ boolean M_Responder (event_t* ev)
 	    {
 	      int value = ptr1->var.def->location->i;
 
-	      if (old_value == -1)
+	      if (!(ptr1->m_flags & S_THERMO) && old_value == -1)
```

Comment:
Hm, I don't understand this change. Why do we have to special case thermometers here?

Reply:
Thermo applies the new value instantly, without confirmation. Therefore, if the user presses ESC while moving the thermo, it does not reset to the default value.